### ISSUE TITLE
plugin converts null position into LAT 0, LON 0 and gives wrong mag variation

### ISSUE THREAD

**Issue by mgrouch** [SignalK/signalk-derived-data#102]
plugin converts null position into LAT 0, LON 0 and gives wrong mag variation
and true heading as the result.

**Comment by msallin** [comment]
@panaaj you're right, this was actually fixed back in 2022 by #103 (commit e33e9b0). The guard in `calcs/magneticVariation.js` already bails out on null position / null lat-long, so null no longer gets converted to 0,0.

While looking at it I did spot an adjacent edge case: the current guard uses truthiness (`!position.latitude || !position.longitude`), which also drops valid positions exactly on the equator or on the Greenwich meridian. Opened #184 with a fix and tests for that.

@panaaj @tkurki safe to close this issue I think.
